### PR TITLE
#6 task. andrei-zhivotkov

### PR DESCRIPTION
**Pull request by Inquisitor1229**

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -270,7 +311,23 @@ function reverseInteger(num) {
...
+    {
+        number = parseInt(ccn.toString().charAt(i), 10);
+
+        if (evenChar && (number *= 2) > 9) number -= 9;
+
+		sum += number;
+        evenChar = !evenChar;
+    }
+
+	return (sum % 10) == 0;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -289,7 +346,24 @@ function isCreditCardNumber(ccn) {
  *   165536 (1+6+5+5+3+6 = 26,  2+6 = 8) => 8
  */
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    let num2 = num,
+        sum = 0;
+
+    while (num2.toString().length != 1) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `!==` is always preferable
sometimes ''!=' may cause an error 
(check at console : 
"0"  != 0   
and  "0" !== 0)

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -289,7 +346,24 @@ function isCreditCardNumber(ccn) {
...
+        sum += num2 % 10;
+        num2 = (num2 - num2 % 10) / 10;
+        if (num2.toString().length == 1) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -52,7 +52,9 @@ function getFizzBuzz(num) {
  */
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js:52]: There is not travis-ci link for PR

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -392,7 +554,48 @@ function toNaryString(num, n) {
...
+        minlng = 0,
+        equal = false;
+
+    minlng = pathes[0].length;
+
+    for (let i = 1; i <=pathes.length-1; i++) {
+        elem1 = pathes[i];
+        elem2 = pathes[i-1];
+
+        if (elem1[0] != elem2[0]) return ''
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  inequality operator (!==) everytime

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -392,7 +554,48 @@ function toNaryString(num, n) {
...
+            if (elem1[1] != elem2[1]) return '/'
+            else  if (pathes[i].length < minlng) minlng = pathes[i].length; 
+    }
+
+    elem1 = pathes[0];
+
+    outer: for (let i = 0; i <= minlng; i++) {
+        for (let j = 1; j <= pathes.length-1; j++) {
+            elem2 = pathes[j];
+            if ( elem1[j] != '/') {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  inequality operator (!==) everytime

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -392,7 +554,48 @@ function toNaryString(num, n) {
...
+                    if ( elem1[i] == elem2[i]) equal = true;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  equality operator (===) everytime

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -392,7 +554,48 @@ function toNaryString(num, n) {
...
+                    else break outer;
+                }
+            else str += '/';
+        }
+        if (equal) str += elem1[i];
+    }
+
+    if ( str[str.length-1] == '/' ) return str
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  equality operator (===) everytime

File: task/06-conditions-n-loops-tasks.js [hunk 9]
```diff
@@ -392,7 +554,48 @@ function toNaryString(num, n) {
...
+    else {
+        for (let i = str.length-1; i >= 0; i--) {
+            if (str[i] != '/') {str = str.slice(0, i);  }
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  inequality operator (!==) everytime

File: task/06-conditions-n-loops-tasks.js [hunk 10]
```diff
@@ -415,7 +618,28 @@ function getCommonDirectoryPath(pathes) {
...
  */
 function getMatrixProduct(m1, m2) {
-    throw new Error('Not implemented');
+    let  m1rows = m1.length,
+         m1cols = m1[0].length,
+         m2rows = m2.length,
+         m2cols = m2[0].length,
+         result = [];
+
+    if (m1cols != m2rows) return false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  inequality operator (!==) everytime

File: task/06-conditions-n-loops-tasks.js [hunk 11]
```diff
@@ -211,7 +229,14 @@ function findFirstSingleChar(str) {
  *
  */
 function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
-    throw new Error('Not implemented');
+    let interval = '';
+
+    isStartIncluded ? interval += '[' : interval += '(';
+    a > b ? interval += b + ', ' + a : interval += a + ', ' + b;
+    isEndIncluded ? interval += ']' : interval += ')';
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js:236]: avoid this code style, if-else is more readable than multiple conditional operators ? :

File: task/06-conditions-n-loops-tasks.js [hunk 12]
```diff
@@ -450,7 +674,26 @@ function getMatrixProduct(m1, m2) {
...
+
+    if (((position[2][2] == position[2][1]) && (position[2][1] == position[2][0])) 
+        || ((position[2][2] == position[1][2]) && (position[1][2] == position[0][2])) 
+        && (position[2][2] != undefined)) 
+        return position[2][2];
+
+    if (((position[1][0] == position[1][1]) && (position[1][1] == position[1][2])) 
+        || ((position[0][1] == position[1][1]) && (position[1][1] == position[2][1])) 
+        || ((position[0][2] == position[1][1]) && (position[1][1] == position[2][0])) 
+        && (position[1][1] != undefined))
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use strict  inequality operator (!==)  (or strict equality '===' respectively)  everytime

**Review by vramaniuk — COMMENTED** [review]
Please, make changes and PR will be closed)

File: task/06-conditions-n-loops-tasks.js (hunk 4 above)
Comment (Inquisitor1229) [reply task/06-conditions-n-loops-tasks.js:52]: https://travis-ci.org/github/Inquisitor1229/js-assignments/branches

**Comment by Inquisitor1229** [comment]
https://travis-ci.org/github/Inquisitor1229/js-assignments/builds/715795659

**Comment by Inquisitor1229** [comment]
> Please, make changes and PR will be closed)

I make, please check the task.
And when can i get task 3(DB)?

**Comment by vramaniuk** [comment]
Ok,  i'll close PR. You will be provided with access to Task3 chat in slack

**Pull request closed without merging**